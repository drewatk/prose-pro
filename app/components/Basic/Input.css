.error {
  font-size: 15px;
  font-family: sans-serif;
  color: #ad1e1a;
}
