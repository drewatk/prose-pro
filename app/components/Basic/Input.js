import React from 'react';

const Input = ({ ...props }) => <input type="text" {...props} />;

export default Input;
