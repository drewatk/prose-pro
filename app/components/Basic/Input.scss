.error {
  font-size: 15px;
  font-family: sans-serif;
  color: var(--error-color);
}

.input {
  background-color: var(--white-color) !important;
  color: var(--dark-color) !important;
}
