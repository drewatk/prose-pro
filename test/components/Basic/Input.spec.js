import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Input from "app/components/Basic/Input";

Enzyme.configure({ adapter: new Adapter() });

describe("Input", () => {
  it("matches snapshot", () => {
    const props = {
      input: {},
      type: "text",
      meta: { touched: false, error: false },
      label: "label"
    };

    const wrapper = shallow(<Input {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches snapshot (with error)", () => {
    const props = {
      input: {},
      type: "text",
      meta: { touched: true, error: true },
      label: "label"
    };

    const wrapper = shallow(<Input {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
