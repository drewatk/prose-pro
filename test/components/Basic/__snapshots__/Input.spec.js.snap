// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot (with error) 1`] = `
<div>
  <InputGroup
    size="sm"
    tag="div"
  >
    <InputGroupAddon
      addonType="prepend"
      tag="div"
    >
      label
    </InputGroupAddon>
    <Input
      type="text"
    />
  </InputGroup>
  <span
    className="error"
  >
    <Component />
  </span>
</div>
`;

exports[`Input matches snapshot 1`] = `
<div>
  <InputGroup
    size="sm"
    tag="div"
  >
    <InputGroupAddon
      addonType="prepend"
      tag="div"
    >
      label
    </InputGroupAddon>
    <Input
      type="text"
    />
  </InputGroup>
</div>
`;
